ddtrace/tracer/option_test.go

```diff
@@ -529,6 +529,14 @@ func TestTracerOptionsDefaults(t *testing.T) {
 			c := tracer.config
 			assert.Equal(t, &url.URL{Scheme: "http", Host: "testhost:3333"}, c.agentURL)
 		})
+
+		t.Run("code-override-full-URL", func(t *testing.T) {
+			t.Setenv("DD_TRACE_AGENT_URL", "https://custom:1234")
+			tracer := newTracer(WithAgentURL("http://testhost:3333"))
+			defer tracer.Stop()
+			c := tracer.config
+			assert.Equal(t, &url.URL{Scheme: "http", Host: "testhost:3333"}, c.agentURL)
+		})
```

Comment:
LGTM. Can we add a test for the failing condition with an unsupported schema? You can use `internal/log.RecordLogger` combined with `ddtrace/tracer.WithLogger` to capture the output log.

An available example is in `TestReportAbandonedSpans`. Just be careful with `RecordLogger.Logs`, as it's asynchronous and you are going to need `assert.Eventually` to wait the logger to register your output. Otherwise, this test could be flaky, as `RecordLogger.Logs` doesn't guarantee that all tests are flushed to its internal buffer on call.